libkbfs/folder_branch_ops.go

```diff
@@ -4710,27 +4694,22 @@ func (fbo *folderBranchOps) rekeyLocked(ctx context.Context,
 			// The rekey bit was already set, and there's nothing else
 			// we can to do, so don't put any new revisions.
 			fbo.log.CDebugf(ctx, "No further rekey possible by this user.")
-			return nil
+			return RekeyResult{
+				DidRekey:      false,
+				NeedsPaperKey: false,
+			}, nil
 		}
 
 		// Rekey incomplete, fallthrough without early exit, to ensure
 		// we write the metadata with any potential changes
 		fbo.log.CDebugf(ctx,
 			"Rekeyed reader devices, but still need writer rekey")
 
-	case NeedOtherRekeyError:
-		stillNeedsRekey = true
-	case NeedSelfRekeyError:
+	case NeedOtherRekeyError, NeedSelfRekeyError:
 		stillNeedsRekey = true
 
 	default:
-		if err == context.DeadlineExceeded {
-			fbo.log.CDebugf(ctx, "Paper key prompt timed out")
-			// Reschedule the prompt in the timeout case.
-			stillNeedsRekey = true
-		} else {
-			return err
-		}
+		return RekeyResult{}, err
```

Comment:
> since other error types might be fatal
hmmm it seems the question here is if we want to stop when there is an error that could be fatal, or keep trying when there is an error that could be non-fatal. We have a TTL now so I was leaning towards the second option, thus this change.